### PR TITLE
chore: svelte-kit-2

### PR DESCRIPTION
**Pull request by jrasm91**
Upgrade to svelte kit 2 and vite 5.
- Redirects no longer need to be thrown
- Type imports are a lot more strict

### CODE REVIEW

**Review by martabal — APPROVED** [review]

File: web/src/lib/components/shared-components/navigation-bar/navigation-bar.svelte [hunk 1]
```diff
@@ -30,7 +30,11 @@
   const logOut = async () => {
     resetSavedUser();
     const { data } = await api.authenticationApi.logout();
-    goto(data.redirectUri || '/auth/login?autoLaunch=0');
+    if (!data.redirectUri.startsWith('/')) {
+      window.location.href = data.redirectUri;
+    } else {
+      goto(data.redirectUri);
+    }
```
Comment (danieldietzler) [inline web/src/lib/components/shared-components/navigation-bar/navigation-bar.svelte:37]: Why is this necessary? Do we sometimes have an URI that's like `auth/login` and `goto` then does not understand that it's still the same host URL?

**Review by danieldietzler — APPROVED** [review]
LGTM! Can we have a linting rule that enforces type imports if we only use the import as a type and not as a value?

File: web/src/lib/components/shared-components/navigation-bar/navigation-bar.svelte (hunk 1 above)
Comment (jrasm91) [reply web/src/lib/components/shared-components/navigation-bar/navigation-bar.svelte:37]: Goto only works for internal urls now

**Review by alextran1502 — APPROVED** [review]
